noir-projects/noir-protocol-circuits/crates/types/src/traits.nr

```diff
@@ -45,12 +45,36 @@ impl ToField for Field {
     }
 }
 
+impl ToField for u1 { fn to_field(self) -> Field { self as Field } }
```

Comment:
Should we add the impl for `boolean` as well? Or maybe when autogenerating serialization code we can check whether it's a primitive type (is it possible?) and then use `as Field` instead of `to_field()`? Same for deserialization.

Reply:
Added. I actually tried to use `to_field()` for everything (for primitive types is essentially a cast), but run into this:

https://github.com/noir-lang/noir/issues/4463

Switched to using the proposed workaround.

Reply:
Scratch that, broke the compiler 🤣 Supported booleans with the old workaround...

Reply:
Scratch the scratch...the compiler is broken somewhere else 💀

Reply:
Fixed